### ISSUE TITLE
Suspended GPU but new workunits are requested

### ISSUE THREAD

**Issue by smoe** [BOINC/boinc#2257]
Hello,
because of my GPU issues (https://github.com/BOINC/boinc/issues/2119) I have suspended GPU computing. But still, tasks also for the GPU are requested. I propose to only download new material for resources that are enabled at the time of the download.
Cheers,
Steffen

```
Thu 23 Nov 2017 11:26:27 PM CET |  | max disk usage: 8.00 GB
Thu 23 Nov 2017 11:26:27 PM CET |  | max CPUs used: 1
Thu 23 Nov 2017 11:26:27 PM CET |  | don't compute while active
Thu 23 Nov 2017 11:26:27 PM CET |  | don't use GPU while active
Thu 23 Nov 2017 11:26:27 PM CET |  | suspend work if non-BOINC CPU load exceeds 25%
Thu 23 Nov 2017 11:26:27 PM CET |  | (to change preferences, visit a project web site or select Preferences in the Manager)
Thu 23 Nov 2017 11:26:31 PM CET | Einstein@Home | work fetch resumed by user
Thu 23 Nov 2017 11:26:36 PM CET | Einstein@Home | Sending scheduler request: To fetch work.
Thu 23 Nov 2017 11:26:36 PM CET | Einstein@Home | Requesting new tasks for CPU and NVIDIA GPU
Thu 23 Nov 2017 11:26:44 PM CET | Einstein@Home | Scheduler request completed: got 6 new tasks
Thu 23 Nov 2017 11:26:46 PM CET | Einstein@Home | Started download of einstein_O1Spot1Lo_1.00_x86_64-pc-linux-gnu__AVX
Thu 23 Nov 2017 11:26:46 PM CET | Einstein@Home | Started download of 20161121_O1MD1_4m_G3473489h_segmentList.seg
Thu 23 Nov 2017 11:26:48 PM CET | Einstein@Home | Finished download of 20161121_O1MD1_4m_G3473489h_segmentList.seg
Thu 23 Nov 2017 11:26:48 PM CET | Einstein@Home | Started download of skygrid_0262.40Hz_O1Spot1.dat
```

**Comment by CharlieFenton** [comment]
I agree this would be useful. because of the "CL file build failure" issue with SETI@home on Mac OAS 10.13, I have GPU computation suspended on one Mac. But it works fine on my other Mac and on my Windows PC, so I don't want to disable it for all my computers.

**Comment by sekerob** [comment]
At least at WCG, you can disable sending work of either GPU or CPU resource type i.e. you can control which project is (dis)allowed to use a resource.

**Comment by CharlieFenton** [comment]
> At least at WCG, you can disable sending work of either GPU or CPU resource type i.e. you can control which project is (dis)allowed to use a resource.

Yes, but not on a per computer basis, unless you designate one computer as "home" and the others as "work" or "school" or something similar. This enhancement request is to allow blocking GPU tasks from all projects for a particular computer locally, without needing to do it on each project's web site.

**Comment by sekerob** [comment]
True, my thought had not extended to farm operators.

**Comment by smoe** [comment]
For many, the BOINC client is the only interface to the scientific project. That is a good thing. Interactions with the project's web site should not be mandatory.

**Comment by sekerob** [comment]
The behavior maybe could be changed, in that if the GPU is suspended on the Activity menu, no work for the GPU is requested on Update, with accompanying message in event log and/or the "No new tasks" button in the BM at left could be split in 2. Would it require boinccmd to be enhanced as well?

**Comment by smoe** [comment]
I have no exact clue how the code is organised, but I would anticipate that piece of logic to be somewhere anchored in the boinc daemon, not in the -manager or -cmd interfaces. What are next steps to address? Should we come up with a patch first (which may then not be accepted)? Or better wait until that feature change is OKed?

**Comment by davidpanderson** [comment]
This is normal behavior - suspending the GPU is assumed to be temporary.

**Comment by sekerob** [comment]
@smoe,

If the issue pertains to a specific machine, the resource can be disabled in cc_config.xml with the <exclude_gpu> option. [url]https://boinc.berkeley.edu/wiki/Client_configuration[/url]. On restart, the resource would no longer be recognized by the client, thus no work be ask for it.

**Comment by smoe** [comment]
@sekerob  Ah, ok, many thanks. I'll leave it for now since the immediate damage (four workunits downloaded that will not be processed until they expire) is already done. This is also since machine is too peripheral for me to be bothered with editing config files. @davidpanderson This does not sound like you tend to accept a patch for it.

**Comment by sirzooro** [comment]
> This is normal behavior - suspending the GPU is assumed to be temporary.
It depends. This "temporary suspend" may last for few months, when people suspend their GPUs during summer time to reduce heat produced. It does not make sense to download WUs during that time, they will timeout.

**Comment by smoe** [comment]
In Northern Europe most energy is produced with water and to use it for heating is common.  I am much with @sirzooro here. Seasonal contributions seem very natural, outside California this is. That said, I still do not get why there are electric heaters that do not also compute :o/
